### PR TITLE
Roidstation Initial Commit

### PR DESCRIPTION
**Pull request by GeneralVeers25**
Snowmap never. Roidstation now.

Adds Roidstation. These are all of the files that are required to implement the latest gimmick map, Roidstation. This still requires heavy playtesting.

Features:
-Map is completely enclosed on one Z-level
-No research outpost, so Xenoarcheologists will really griff the entire station with their deadly artifacts
-Bagel gulag may actually be used now
-It still came out before datums

For a full render please see:
http://mapdiffbot.dextraspace.net/Files/10441188/11040236/0/after.png
May be slightly out of date.

🆑 
* rscadd: added Roidstation.

### CODE REVIEW

**Comment by brndd** [comment]
what is git squashing

t. guy who was in charge of squashing the gits

**Comment by BurntDevil** [comment]
we roid now bois

**Comment by IzayaChan** [comment]
merge now pls

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Do you have a render not for ants

**Comment by aceedex** [comment]
https://media.giphy.com/media/rl0FOxdz7CcxO/giphy.gif

**Comment by Kurfursten** [comment]
As someone who has participated in playtesting and downloaded the map for close examination, I can say that Roidstation tries a lot of new things that could shake up playstyles. It's also not without weaknesses, but it has come a long way. There are some decisions - like the "spider room" in engineering (yes it is what it sounds like) that will probably have to be trial-by-fired but prepare for Duny maintenance bear memes. Honestly I do fear the map getting a stigma after dumb engineering teams get wiped out repeatedly by that chamber. Other things are perhaps more fundamental - the choice on how to orient the upload/core, the already-discussed dangerous proximity of the holodeck to the brig hall, and in some cases lack of clarity as to what qualifies as "radiation-shielded" maintenance... well, we will see how people take to it in the coming weeks.

I recommend putting it on the test server, not that most people will test it.

**Comment by GeneralVeers25** [comment]
@I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE  Yes, we do now. The full render is here: http://mapdiffbot.dextraspace.net/Files/10441188/10827892/0/after.png

Or if you want a smaller, labeled render you can also refer to https://lambda.sx/vZ5.png

**Comment by aceedex** [comment]
Looking at the render, I dont see a way for the asteroid people to get to Z5. I understand that maybe the point is to just have everyone in the one Z level, but I'd wager it might also be beneficial to have a mining outpost on Z5 like most other maps in case of things like Z-level wide artifacts

**Comment by BurntDevil** [comment]
Artifact fuckery one z1 is one of the more interesting experiments of roid tbh

**Comment by SonixApache** [comment]
I can't wait for the taxi-style cell-drain / constant explosion memes.

**Comment by brndd** [comment]
>Looking at the render, I dont see a way for the asteroid people to get to Z5.

I'm not sure I understand what you mean. You can get to Z5 by just... flying there. It's the leviathan roid from (I think) Taxi. Adding shuttle docks and cards to get to these areas was suggested in the thread, and has been added to the list. That reminds me, I'm not sure how supermatter cascade works with no research outpost.

File: code/modules/mining/mine_turfs.dm [hunk 1]
```diff
@@ -547,6 +560,108 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
...
+	switch(severity)
+		if(3.0)
+			return
+		if(2.0)
+			if (prob(70))
+				gets_dug()
+		if(1.0)
+			gets_dug()
+
+/turf/simulated/floor/asteroid/attackby(obj/item/weapon/W as obj, mob/user as mob)
```
Comment (ShiftyRail) [inline code/modules/mining/mine_turfs.dm:627]: `as obj`, `as mob` are not needed

File: code/modules/mining/mine_turfs.dm [hunk 2]
```diff
@@ -547,6 +560,108 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
...
+			to_chat(user, "<span class='rose'>This area has already been dug.</span>")
+			return
+		to_chat(user, "<span class='rose'>You start digging.<span>")
+		playsound(src, 'sound/effects/rustle1.ogg', 50, 1) //russle sounds sounded better
+		if(do_after(user, src, used_digging.digspeed) && user) //the better the drill, the faster the digging
+			playsound(src, 'sound/items/shovel.ogg', 50, 1)
+			to_chat(user, "<span class='notice'>You dug a hole.</span>")
+			gets_dug()
+	else
+		..(W,user)
```
Comment (ShiftyRail) [inline code/modules/mining/mine_turfs.dm:647]: The arguments are passed implicitly

Comment (brndd) [reply code/modules/mining/mine_turfs.dm:647]: Everything about this turf is copypasted from the existing unsimulated asteroid floor

**Comment by aceedex** [comment]
I meant a way to reliably get there without spacing yourself and doing the little dance at the edge of the Z level or with telesci, like a shuttle. but if thats already under consideration its all well and good

**Comment by GeneralVeers25** [comment]
The Salvage Shuttle already has a dock that will go there. I don't know how to add shuttle cards like the Mecha Graveyard may have, but that would definitely be a good option.

**Comment by Kurfursten** [comment]
Where's the teleporter?

**Comment by brndd** [comment]
>Where's the teleporter?

Next to Escape and the Pod Bay. "Telescience" is merged with the paramedic station for memes.

**Comment by brndd** [comment]
Oh also the PR branch has been unfucked and properly rebased, so testing the map by directly pulling the roidstation-release branch should work now. Here is a direct zip download: https://github.com/GeneralVeers25/vgstation13/archive/roidstation-release.zip

**Comment by brndd** [comment]
And with that, it seems everyone involved has slipped in whatever final fixes they wanted to do. Should be ready to merge as far as we're concerned.

**Comment by jknpj** [comment]
It's time.

**Pull request merged**